docs/markdown.rst

```diff
@@ -2,56 +2,106 @@ markdown
-in the file are not overwritten, so you may use the ``markdown`` command
-multiple times to create a single document containing the text from multiple
-screens, like thoughts from several dwarves or descriptions from multiple
-artifacts.
-
-The screens which have been tested and known to function properly with this
-script are:
-
-#. dwarf/unit 'thoughts' screen
-#. item/art 'description' screen
-#. individual 'historical item/figure' screens
-#. manual pages
-#. announcements screen
-#. combat reports screen
-#. latest news (when meeting with liaison)
-
-There may be other screens to which the script applies. It should be safe to
-attempt running the script with any screen active. An error message will inform
-you when the selected screen is not appropriate for this script.
+    :summary: Save descriptions of selected units/items in markdown format (e.g., for Reddit).
+    :tags: fort | inspection | units
+
+Saves descriptions of selected units or items to a markdown file.
+
+
+For units, the script retrieves:
+#. Name, race, age, profession
+#. Description from the Unit/Health/Description screen
+#. Traits from the Unit/Personality/Traits screen
+The script works for all units with text in the Description &/or Personality/Traits tabs, 
+such as dwarves, dogs, elves, goblins, and beasts.
+
+For items, it retrieves:
+#. Decorated name (e.g., "☼«☼Chalk Statue of Dakas☼»☼")
+#. Full description from the item's view sheet
+The script works for most items with in-game descriptions and names, including those in storage,
+on the ground, installed as a building, or worn/carried by units.
+
+The script appends a markdown-formatted version of the text to a target file 
+for easy sharing, e.g., on Reddit.
+
+By default, entries are appended, not overwritten, allowing the ``markdown`` command 
+to compile descriptions of multiple items & units in a single document.
+
+By default, data is stored in markdown_{YourWorldName}_export.md.
+
+See `forum-dwarves` for BBCode export (e.g., for the Bay12 Forums).
+
 
 Usage
 -----
 
 ::
 
-    markdown [-n] [<name>]
+    markdown [-o] [<filename>]
+
+Appends the description of the selected unit/item 
+to ``markdown_{world_name}_export.md`` file by default. 
+Specifying a filename will append to ``markdown_{filename}.md`` instead,
+which is useful for organizing data by category or topic.
+The [-o] argument tells the script to overwrite the output file.
+
+Options
+-------
 
-The output is appended to the ``md_export.md`` file by default. If an alternate
-name is specified, then a file named like ``md_{name}.md`` is used instead.
+``-o``
+    Overwrite the output file, deleting previous entries.
+``help`` 
+    Show help.
 
 Examples
 --------
 
-``markdown``
-    Appends the contents of the current screen to the ``md_export.md`` file.
-``markdown artifacts``
-    Appends the contents of the current screen to the ``md_artifacts.md`` file.
+::
 
-Options
--------
+    markdown
```

Comment:
It seems like this is a tool that could benefit from a global hotkey - (`Ctrl-Shift-s`?) so people can click on things with the mouse and export them quickly with their left hand. This would be added to the keybindings file in the other repo: https://github.com/DFHack/dfhack/blob/develop/data/init/dfhack.keybindings.init

Reply:
I'll add it later, thank you

Reply:
Is it correct? 
``# save description of selected unit or item in markdown format 
keybinding add Ctrl-Shift-S@dwarfmode markdown``

Reply:
that refers to the "context" where the hotkey should be considered active. `dwarfmode` is fort mode, but you can make the hotkey local to more specific contexts (the strings returned from `dfhack.gui.getCurFocus(true)`). that way, the hotkey will only show up in the DFHack context menu where it is applicable.

e.g.

```
keybinding add Ctrl-Shift-S@dwarfmode/ViewSheets/UNIT/Health/Description|dwarfmode/ViewSheets/UNIT/Personality/Traits|...
```

Reply:
Same, let's add it later

Reply:
When I call the command with the hotkey, it doesn't show success message (or any message). Is it expected behavior?

Reply:
I suggest using Shift+R. 
The conflicts I see are:
Ctrl+R - recording
(easy to misclick)

and

# materials: q->workshop; b->select items:
#keybinding add Shift-R "job-material RHYOLITE"
Which is fine, because we will only use the command in our context window